EIPS/eip-1193.md

```diff
@@ -7,104 +7,183 @@ status: Draft
+
+### send (DEPRECATED)
 
-See [available JSON-RPC methods](https://github.com/ethereumproject/go-ethereum/wiki/JSON-RPC#json-rpc-methods).
+Due to conflicting implementations and specifications, this method is unreliable and should not be used.
+
+```typescript
+ethereum.send(...args: Array<any>): unknown;
+```
 
 ### Events
 
-Events are emitted using [EventEmitter](https://nodejs.org/api/events.html).
+Events follow the [Node.js `EventEmitter`](https://nodejs.org/api/events.html) API.
 
 #### notification
 
-All subscriptions from the node emit on notification. Attach listeners with:
+The Provider emits `notification` upon receipt of a notification.
+Notifications may include JSON-RPC notifications, GraphQL subscriptions, or any other event that the Provider decides to treat as a notification.
+
+```typescript
+interface ProviderNotification {
+  type: string;
+  data: unknown;
+}
 
-```js
-ethereum.on('notification', listener: (result: any) => void): this;
+ethereum.on('notification', listener: (notification: ProviderNotification) => void): ethereum;
 ```
 
-To create a subscription, call `ethereum.send('eth_subscribe', [])` or `ethereum.send('shh_subscribe', [])`.
+##### Subscriptions
 
-See the [eth subscription methods](https://github.com/ethereum/go-ethereum/wiki/RPC-PUB-SUB#supported-subscriptions) and [shh subscription methods](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_subscribe).
+Some clients like [Geth](https://geth.ethereum.org/docs/rpc/pubsub) and [Parity](https://wiki.parity.io/JSONRPC-eth_pubsub-module) support Publish-Subscribe (_Pub-Sub_) using JSON-RPC notifications. This lets you subscribe and wait for events instead of polling for them.
+See the [`eth_` subscription methods](https://geth.ethereum.org/docs/rpc/pubsub) and [`shh_` subscription methods](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_subscribe) for details.
 
 #### connect
 
-The provider emits `connect` on connect to a network.
+The Provider emits `connect` when it becomes connected. This includes:
 
-```js
-ethereum.on('connect', listener: () => void): this;
-```
+- when the Provider first connects to a chain after being initialized
+- when the Provider connects to a chain, after the `close` event was emitted
 
-You can detect which chain by sending `eth_chainId`:
+```typescript
+interface ProviderConnectInfo {
+  chainId: string;
+  [key: string]: unknown;
+}
 
-```js
-const chainId = await ethereum.send('eth_chainId');
-> 1
+ethereum.on('connect', listener: (connectInfo: ProviderConnectInfo) => void): ethereum;
 ```
 
+The event emits an object with a hexadecimal string `chainId` per the `eth_chainId` Ethereum RPC method, and other properties as determined by the Provider.
+
 #### close
 
-The provider emits `close` on disconnect from a network.
+The Provider emits `close` when it becomes disconnected from all chains.
```

Comment:
Is there a way to know when the provider becomes disconnected from _a particular_ chain?

Reply:
Yes, by knowing which chain you were connected to at the time `close` was emitted. This spec inherits the assumption of `net_version` and `eth_chainId` that a provider/client is only connected to a single chain at any one time.

Reply:
The "all chains" wording here suggests that there could be multiple simultaneous connections, no?